tools/update_tool

```diff
@@ -0,0 +1,439 @@
+            ["git", "show", f"HEAD^:{file}"],
+            capture_output=True,
+            check=True)
+        return json.loads(command.stdout)
+    except subprocess.CalledProcessError:
+        return None
+    return None
+
+
+def get_db_entry_at_current_version(file):
+    """
+    Retrieves a version of a DB file at the current state of the DB
+    """
+    if not file:
+        return None
+    with open(file, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+def filter_fn(imi):
+    """
+    Filter specific fields in the image-info that can't be compared
+    together.
+    """
+    def lvm2(imi):
+        """
+        LVM2 partitions have a UUID that is not fixed. Replace the value
+        upon comparison time
+        """
+        partitions = imi.get("partitions")
+        if partitions:
+            for partition in partitions:
+                if partition.get("fstype") == "LVM2_member":
+                    partition["uuid"] = "2022-07-01-fixed-uuid"
+        return imi
+
+    def iso(imi):
+        """
+        For isos, the partition UUID is the date of the build. Replace
+        that with a fixed one for the comparison.
+        """
+        if "image-format" in imi and "type" in imi["image-format"] and imi["image-format"]["type"] == "raw":
+            if "partitions" in imi:
+                for partition in imi["partitions"]:
+                    if "fstype" in partition:
+                        if partition["fstype"] == "iso9660":
+                            partition["uuid"] = "2022-07-01-fixed-uuid"
+        return imi
+    return iso(lvm2(imi))
+
+
+def diff(what, json1, json2, tool="vimdiff"):
+    """
+    Starts a diff between two json objects with a specific tool to do so. Writes
+    each json to disk in a temporary directory and then invokes the difftool on
+    these two files.
+    """
+    with tempfile.TemporaryDirectory() as tempdir:
+        a = os.path.join(tempdir, "1")
+        b = os.path.join(tempdir, "2")
+        with open(a, encoding="utf-8", mode="w") as f:
+            json.dump(json1, f, indent=4)
+        with open(b, encoding="utf-8", mode="w") as f:
+            json.dump(json2, f, indent=4)
+        opendiff = input(f"-   diff {what}? [Y, n]")
+        if opendiff in ("y", "Y", "", None):
+            subprocess.call([tool,  a, b])
+
+
+def extract(db_entry, do_filter=True):
+    """
+    Returns two json objects from a db entry. One manifest and one image info.
+    Can apply (and will by default) filters to the image-info to get rid of some
+    volatile UUIds that aren't important.
+    """
+    manifest = None
+    imi = None
+    if not db_entry:
+        return None, None
+    if db_entry["image-info"]:
+        if do_filter:
+            imi = filter_fn(db_entry["image-info"])
+        else:
+            imi = db_entry["image-info"]
+    if db_entry["manifest"]:
+        manifest = db_entry["manifest"]
+    return manifest, imi
+
+
+def get_symbol(label):
```

Comment:
Oh, I didn't see this before. I think this is unused now, right? Can you remove it?

Reply:
Thanks for catching that!